### PR TITLE
test for alternative success exit codes

### PR DESCRIPTION
**Pull request by mr-c**
Initial fix for #627

### CODE REVIEW

**Comment by anton-khodak** [comment]
cwltool currently skips successCodes if the code is 2: https://github.com/common-workflow-language/cwltool/blob/52c276d1a384bed977f2355513f1c07a66b11971/cwltool/job.py#L246, needs a fix

**Comment by anton-khodak** [comment]
should you add it to v1.1 as well?

**Comment by mr-c** [comment]
> cwltool currently ignores successCodes if an OSError happens due to missing files: https://github.com/common-workflow-language/cwltool/blob/52c276d1a384bed977f2355513f1c07a66b11971/cwltool/job.py#L246, needs a fix

True, that should reference `errno.ENOENT` instead of `2` but I think that is correct to ignore `successCodes` here as the command was never run and there is no exit code to examine.

**Comment by mr-c** [comment]
> should you add it to v1.1 as well?
We'll do another sync of v1.0 to v1.1 later

**Comment by anton-khodak** [comment]
so you should change the test so that shell command is actually executed, returns a non-zero code, and successCode is actually checked

File: v1.0/v1.0/exit-success.cwl [hunk 1]
```diff
@@ -0,0 +1,11 @@
+#!/usr/bin/env cwl-runner
+cwlVersion: v1.0
+class: CommandLineTool
+
+inputs: []
+baseCommand: /bin/false
```
Comment (anton-khodak) [inline v1.0/v1.0/exit-success.cwl]: can we change it to just `"false"`? it is /usr/bin/false on MacOS

Comment (mr-c) [reply v1.0/v1.0/exit-success.cwl]: Sure

**Comment by mr-c** [comment]
Thank you @anton-khodak !

**Pull request merged**